src/compiler/failsafe.h

```diff
@@ -0,0 +1,32 @@
+/*!
+ * Copyright (c) 2021 by Contributors
+ * \file failsafe.h
+ * \brief C code generator (fail-safe). The generated code will mimic prediction logic found in
+ *        XGBoost
+ * \author Hyunsu Cho
+ */
+
+#ifndef TREELITE_COMPILER_FAILSAFE_H_
+#define TREELITE_COMPILER_FAILSAFE_H_
+
+#include <treelite/compiler.h>
+#include <treelite/compiler_param.h>
+#include <memory>
+
+namespace treelite {
+namespace compiler {
+
+class FailSafeCompilerImpl;
+
+class FailSafeCompiler : public Compiler {
+ public:
+  explicit FailSafeCompiler(const CompilerParam& param);
+  CompiledModel Compile(const Model& model) override;
+ private:
+  std::unique_ptr<FailSafeCompilerImpl> pimpl_;
+};
```

Comment:
Interesting, I always find using unique ptr with incomplete class troublesome due to missing destructor.  Not sure how did you get it to work.

Reply:
Take a look at https://www.fluentcpp.com/2017/09/22/make-pimpl-using-unique_ptr/. The key is to explicitly define the constructor for `FailSafeCompiler`.

Reply:
Thanks for the reference.  That's a weird aspect of c++.